client/js/app/stores/ExplorerStore.js

```diff
@@ -84,6 +101,87 @@ function _getDefaultFilterCoercionType(explorer, filter) {
 }
 
 /**
+ * Runs through the changeset and moves data around if necessary
+ * @param {Object} explorer   The explorer model that is being updated
+ * @param {Object} updates    The updated explorer model
+ * @return {Object}           The new set of updates
+ */
+function _prepareUpdates(explorer, updates) {
+  var newModel = _.assign({}, explorer, updates);
+
+  newModel = _removeEmailExtractionFields(explorer, newModel);
+  newModel = _migrateFunnelSteps(explorer, newModel);
+
+  return newModel;
+}
+
+/**
+ * If the query got changed to a funnel, move the step-specific parameters to a steps object.
+ * and vice versa if it got changed FROM a funnel
+ * @param {Object} explorer The explorer model that is being updated
+ * @param {Object} newModel The updated explorer model
+ * @return {Object}         The new set of updates
+ */
+function _migrateFunnelSteps(explorer, newModel) {
+  var sharedProperties = ['event_collection', 'time', 'timezone', 'filters']
+  if(newModel.query.analysis_type === 'funnel' && explorer.query.analysis_type !== 'funnel') {
+    // Changing TO funnels
+    var firstStep = _defaultStep();
+    firstStep.active = true;
+
+    _.each(sharedProperties, function (key) {
+      if(typeof(explorer.query[key]) !== 'undefined' && explorer.query[key] !== null) {
+        firstStep[key] = explorer.query[key] 
+      }      
+
+      delete newModel.query[key]
+    });
+
+    if(typeof(explorer.query.target_property) !== 'undefined' && explorer.query.target_property !== null) {
+      firstStep.actor_property = explorer.query.target_property;
+      delete explorer.query.target_property;
+    }
+
+    newModel.query.steps = [firstStep];
+
+  } else if (newModel.query.analysis_type !== 'funnel' && explorer.query.analysis_type === 'funnel') {
```

Comment:
I'm thinking maybe this should be a separate function, instead of an `else if`...  thoughts?

I wrote it as one, because the logic is all so similar... but it's a big messy func.